### ISSUE TITLE
Create new Critical Flows for Progressive Onboarding

### ISSUE THREAD

**Issue by dmallory42** [Automattic/woocommerce-payments#6213]
### Description

We need to add a new [critical flow](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) for the new Onboarding flow prior to launch. 

### Acceptance criteria

* A critical flow is added covering the behaviour of the new onboarding flow.
* The flow is tested prior to launch.

**Comment by brandonwanA8C** [comment]
Same questions/recommendations as [6214](https://github.com/Automattic/woocommerce-payments/issues/6214).
Sounds like this one is more critical & foundational than 6214.

**Comment by daquinons** [comment]
Increasing the scope to 3 to take into consideration what was added https://github.com/Automattic/woocommerce-payments/issues/6214#issuecomment-1748785128

**Comment by vladolaru** [comment]
After learning about the intention behind critical flows (focusing on only critical, non-recoverable, non-work-around flows), I've decided to create **only one critical flow for Progressive Onboarding** that excludes the builder-path and non-PO accounts. We only want to know that when a merchant is eligible for PO, the flow works. If the _builder path_ doesn't work, it is not critical.

The rest of the onboarding-related flows are covered by other critical flows that _probably should be updated,_ especially once we have GA for the new onboarding wizard.

**The testing instructions** for the new `Progressive Onboard via Payments Connect` critical flow [are here](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#progressive-onboard-via-payments-connect).
I've added the new `Progressive Onboard via Payments Connect` critical flow to the [WooPayments critical flows table](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and linked it in the testing instructions page [merchant TOC](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#merchant).
